instrumentation/resources/library/src/test/java/io/opentelemetry/instrumentation/resources/CgroupV2ContainerIdExtractorTest.java

```diff
@@ -30,39 +33,42 @@ void fileNotReadable() {
-        "468 456 0:149 / /sys ro,nosuid,nodev,noexec,relatime - sysfs sysfs ro",
-        "469 468 0:30 / /sys/fs/cgroup ro,nosuid,nodev,noexec,relatime - cgroup2 cgroup rw",
-        "470 466 0:145 / /dev/mqueue rw,nosuid,nodev,noexec,relatime - mqueue mqueue rw",
-        "471 466 0:150 / /dev/shm rw,nosuid,nodev,noexec,relatime - tmpfs shm rw,size=65536k",
-        "472 456 254:1 /docker/containers/dc64b5743252dbaef6e30521c34d6bbd1620c8ce65bdb7bf9e7143b61bb5b183/resolv.conf /etc/resolv.conf rw,relatime - ext4 /dev/vda1 rw",
-        "473 456 254:1 /docker/containers/dc64b5743252dbaef6e30521c34d6bbd1620c8ce65bdb7bf9e7143b61bb5b183/hostname /etc/hostname rw,relatime - ext4 /dev/vda1 rw",
-        "474 456 254:1 /docker/containers/dc64b5743252dbaef6e30521c34d6bbd1620c8ce65bdb7bf9e7143b61bb5b183/hosts /etc/hosts rw,relatime - ext4 /dev/vda1 rw",
-        "376 466 0:148 /0 /dev/console rw,nosuid,noexec,relatime - devpts devpts rw,gid=5,mode=620,ptmxmode=666",
-        "377 457 0:146 /bus /proc/bus ro,nosuid,nodev,noexec,relatime - proc proc rw",
-        "378 457 0:146 /fs /proc/fs ro,nosuid,nodev,noexec,relatime - proc proc rw",
-        "379 457 0:146 /irq /proc/irq ro,nosuid,nodev,noexec,relatime - proc proc rw",
-        "380 457 0:146 /sys /proc/sys ro,nosuid,nodev,noexec,relatime - proc proc rw",
-        "381 457 0:146 /sysrq-trigger /proc/sysrq-trigger ro,nosuid,nodev,noexec,relatime - proc proc rw",
-        "382 457 0:151 / /proc/acpi ro,relatime - tmpfs tmpfs ro",
-        "383 457 0:147 /null /proc/kcore rw,nosuid - tmpfs tmpfs rw,size=65536k,mode=755",
-        "384 457 0:147 /null /proc/keys rw,nosuid - tmpfs tmpfs rw,size=65536k,mode=755",
-        "385 457 0:147 /null /proc/timer_list rw,nosuid - tmpfs tmpfs rw,size=65536k,mode=755",
-        "386 468 0:152 / /sys/firmware ro,relatime - tmpfs tmpfs ");
+  @Test
+  void extractSuccess_podman() throws Exception {
+    when(filesystem.isReadable(V2_CGROUP_PATH)).thenReturn(true);
+    Stream<String> fileContent = getTestPodmanFileContent();
+    when(filesystem.lines(V2_CGROUP_PATH)).thenReturn(fileContent);
+    CgroupV2ContainerIdExtractor extractor = new CgroupV2ContainerIdExtractor(filesystem);
+    Optional<String> result = extractor.extractContainerId();
+    assertThat(result.orElse("fail"))
+        .isEqualTo("2a33efc76e519c137fe6093179653788bed6162d4a15e5131c8e835c968afbe6");
+  }
+
+  private static Stream<String> getTestDockerFileContent() throws Exception {
+    return fileToStreamOfLines("docker_proc_self_mountinfo");
+  }
+
+  private static Stream<String> getTestPodmanFileContent() throws Exception {
+    return fileToStreamOfLines("podman_proc_self_mountinfo");
+  }
+
+  private static Stream<String> fileToStreamOfLines(String filename) throws IOException {
+    try (InputStream in =
+        CgroupV2ContainerIdExtractorTest.class.getClassLoader().getResourceAsStream(filename)) {
+      byte[] buff = new byte[100 * 1024];
+      int rc = in.read(buff);
```

Comment:
I don't think `read` is guaranteed to read as many bytes as possible (I've seen people implement `readFully` on top of it when that is needed)

how about BufferedReader.lines()?

Reply:
Yeah that's much nicer, thanks!